library/SmartyPlugins/function.community_chooser.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ * @package vanilla-smarty
+ * @since 2.0
+ */
+
+/**
+ * Create a subcommunity chooser placeholder.
+ *
+ * If subcommunities is enabled, this will be overwritten by the chooser.
+ *
+ * @param array $params The options for the component if it exists.
+ * - buttonType: 'primary', 'standard', 'titleBarLink', 'transparent', 'transluscent'
+ * - fullWidth: Whether or not the chooser button shoudl take the full width of it's container.
+ *
+ * @return string Some HTML.
+ */
+function smarty_function_community_chooser(array $params): string {
+    $props = json_encode($params);
+    return "<span data-react='subcommunity-chooser' data-props='$props'></span>";
```

Comment:
Couldn't we remove this html out of here to react component somehow?

Reply:
We can't. This is what the react component actually mounts on top of (The `data-react` is determines what registered component is mounted).

The `data-props` are what get passed to the component. You can see the registration in the frontend here. https://github.com/vanilla/multisite/blob/master/plugins/subcommunities/src/scripts/entries/forum.tsx#L9

If that isn't registered (eg. subcommunities not enabled) then this is just an empty span tag.

Reply:
I would make it a twig view, but the utility I would need to render it in a global function is sitting in my other PR.

Reply:
To me it is not even about twig but about react. But up to you. I just feel this is dirty :-)